flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumerTest.java

```diff
@@ -1062,4 +1104,28 @@ public boolean isClearCalled() {
 
 		return fakeRestoredState;
 	}
+
+	private final class NonSerializableDeserializationSchema implements KinesisDeserializationSchema<String> {
```

Comment:
Can you please add a comment explaining why this is not serializable and why the following one is serializable? Basically pointing out this class is not static.